components/create_team_restricted_modal/create_team_restricted_modal.tsx

```diff
@@ -0,0 +1,108 @@
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch<DispatchFunc>();
+
+    const subscription = useSelector(selectCloudSubscription);
+    const isSystemAdmin = useSelector(isCurrentUserSystemAdmin);
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.CREATE_TEAM_RESTRICTED_MODAL));
+
+    if (!show) {
+        return null;
+    }
+
+    const hadPrevCloudTrial = subscription?.is_free_trial === 'false' && subscription?.trial_end_at > 0;
+    const showStartTrial = isSystemAdmin && !hadPrevCloudTrial;
+
+    const dismissAction = async () => {
+        await dispatch(closeModal(ModalIdentifiers.CREATE_TEAM_RESTRICTED_MODAL));
+    };
+
+    const openPricingModal = async () => {
+        await dismissAction();
+
+        await dispatch(openModal({
+            modalId: ModalIdentifiers.PRICING_MODAL,
+            dialogType: PricingModal,
+        }));
+    };
+
+    return (
+        <GenericModal
+            id='CreateTeamRestrictedModal'
+            className='CreateTeamRestrictedModal'
+            useCompassDesign={true}
+            modalHeaderText={hadPrevCloudTrial ? (
+                formatMessage({id: 'create_team_restricted_modal.title.prevTrial', defaultMessage: 'Upgrade to create unlimited teams'})
+            ) : (
+                formatMessage({id: 'create_team_restricted_modal.title', defaultMessage: 'Try unlimited teams with a free trial'})
+            )}
+            onExited={dismissAction}
+        >
+            <div className='CreateTeamRestrictedModal__body'>
+                <p className='CreateTeamRestrictedModal__description'>
+                    {hadPrevCloudTrial ? (
+                        formatMessage({
+                            id: 'create_team_restricted_modal.description.prevTrial',
+                            defaultMessage: 'Multiple teams allow for context-specific spaces that are more attuned to your and your teams’ needs. Upgrade to the Professional plan to create unlimited teams.',
+                        })
+                    ) : (
+                        formatMessage({
+                            id: 'create_team_restricted_modal.description',
+                            defaultMessage: 'Create unlimited teams with one of our paid plans. Get the full experience of Enterprise when you start a free, 30 day trial.',
+                        })
+                    )}
+                </p>
+                {!hadPrevCloudTrial && (
+                    <p className='CreateTeamRestrictedModal__terms'>
+                        <FormattedMarkdownMessage
+                            id='create_team_restricted_modal.agreement'
+                            defaultMessage='By selecting **Try free for 30 days**, I agree to the [Mattermost Software Evaluation Agreement](!https://mattermost.com/software-evaluation-agreement/), [Privacy Policy](!https://mattermost.com/privacy-policy/), and receiving product emails.'
```

Comment:
Although existing usages of this link do exactly this, could we extract out the software evaluation link to [`LicenseLinks`](https://github.com/mattermost/mattermost-webapp/blob/master/utils/constants.tsx#L908) and reuse `AboutLinks.PRIVACY_POLICY`?  Example of embedding the link in a `FormattedMessage` : https://github.com/mattermost/mattermost-webapp/blob/master/components/collapsed_reply_threads_beta_modal/collapsed_reply_threads_beta_modal.tsx#L44-L59

There's two main reasons for this:
* We've deprecated `FormattedMarkdownMessage` because it does some internationalization things wrong and is more powerful than we need or want (a hypothetical malicious translator can inject bad code)
* It lessens the burden on translators initially and when we update links.